### PR TITLE
Allow to use FQCN and strings names in filters

### LINKED ISSUE #5781: Auto quess filter type broken in sonata-project/admin-bundle 3.56.0
After update 

`- Updating sonata-project/admin-bundle (3.55.0 => 3.56.0): Loading from cache`

all filters with null type throw error

`No attached service to type named `doctrine_orm_number``

here is line 
`$datagridMapper->add('id')`

id definition 

```
  /**
     * @ORM\Column(type="integer")
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="AUTO")
     */
    protected $id;

```

Rolling back to 3.55.0 fix errors

### PR DESCRIPTION
**Pull request by franmomu**
## Subject

Since I removed that piece of code and the ORM/MongoDB/PHPCR are not updated accordingly, I think the best solution is to restore the ability of using string names. I'll try to trigger deprecations when using them and then remove completely in next major version.

I am targeting this branch, because these changes are BC.

Closes #5781

## Changelog

```markdown
### Fixed
- Ability of using string names and FQCNs to define filter types
```

### CODE REVIEW

File: src/DependencyInjection/Compiler/AddFilterTypeCompilerPass.php [hunk 1]
```diff
@@ -35,6 +35,11 @@ public function process(ContainerBuilder $container)
             $serviceDefinition->setPublic(true); // Temporary fix until we can support service locators
 
             $types[$serviceDefinition->getClass()] = $id;
+
+            // NEXT_MAJOR: Remove the alias when dropping support for symfony 2.x
```
Comment (core23) [inline src/DependencyInjection/Compiler/AddFilterTypeCompilerPass.php]: This is not related to symfony 2 anymore, it's related to the AdminDoctrineOrmBundle version

Comment (core23) [reply src/DependencyInjection/Compiler/AddFilterTypeCompilerPass.php]: Adding a conflicting version should solve this even better

Comment (franmomu) [reply src/DependencyInjection/Compiler/AddFilterTypeCompilerPass.php]: The problem is if users are still using string names like `doctrine_orm_*`, it won't work
